python/darpinstances/solution_checker.py

```diff
@@ -160,24 +166,52 @@ def check_plan(plan: VehiclePlan, plan_counter: int, instance: DARPInstance, use
 
         equipment = action_data.action.request.equipment
         if equipment != 0:
-            if action_data.action.action_type == ActionType.PICKUP:
+            if is_pickup:
                 if not any(equipment in config for config in available_configurations):
                     print("Equipment {} not available in vehicle equipment list.".format(equipment))
                     plan_ok = False
                 used_equipment.append(equipment)
-            elif action_data.action.action_type == ActionType.DROP_OFF:
+            elif is_drop_off:
                 used_equipment.remove(equipment)
 
         cost += travel_time
 
+        # operation time check
+        start_time = plan.vehicle.operation_start
+        end_time = plan.vehicle.operation_end
+        if (start_time>0 & end_time>0 & (action_data.departure_time < start_time) | (action_data.departure_time > end_time)):
+            print("Action is served outside of operation times.".format(equipment))
+            plan_ok = False
+
+        # vehicle id check
+        if action_data.action.request.vehicle_id != 0:
+            if action_data.action.request.vehicle_id != vehicle_index:
+                print("Request {} is not for vehicle {}.".format(action_data.action.request.index, vehicle_index))
+                plan_ok = False
+
+
+        # pause check
+        is_vehicle_empty = free_capacity == plan.vehicle.capacity  if not used_equipment else False
+        driving_duration = time - driving_start
+        if is_pickup & is_vehicle_empty:
+            if (time - pause_start) >= min_pause_length:
+                driving_start = time - travel_time
+                driving_duration = travel_time
+        elif is_drop_off & is_vehicle_empty:
+            pause_start = time
```

Comment:
this logic is not correct in my opinion. The pause can happen on both pickup and drop off actions. You need only one counter (driving start), and the pause length is equal to the diff between time and min time (see waiting to min time)